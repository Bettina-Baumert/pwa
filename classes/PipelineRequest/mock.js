/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
/**
 * Mocked PipelineRequest.
 * Use it any time you need to mock a PipelineQuest.
 *
 * Usage:
 * For a test usually you need to have control over how the dispatch resolves.
 * For this purpose, use .setMockedDispatchResolver to set up a callback which will be called
 * after PipelineRequest.dispatch is called instead of the original one.
 *
 * To make it work, you can mock the original PipelineRequest by using jest.mock
 * after your test import statements. For example:
 * jest.mock(
 *  '@shopgate/pwa-core/classes/PipelineRequest',
 *  () => (
 *    mockedPipelineRequestFactory((mockInstance, resolve, reject) => {
 *      // This is a callback which will be executed in .dispatch.
 *    })
 *  )
 * );
 */
class MockedPipelineRequest {
  /**
   * Getter for mockedDispatchResolver which is additional helper function for custom, mock-only
   * .dispatch resolver.
   * @return {function()}
   */
  static get mockedDispatchResolver() {
    return () => {};
  }

  /**
   * @inheritDoc
   */
  constructor(name) {
    this.name = name;
    this.input = {};
  }

  /**
   * @inheritDoc {PipelineRequest}
   */
  setInput(mockedInput = {}) {
    this.input = mockedInput;
    return this;
  }

  /**
   * Returns promise and calls .mockedDispatchResolver which can be set up by using
   * .setMockedDispatchResolver.
   * @return {Promise}
   */
  dispatch() {
    return new Promise((resolve, reject) => {
      this.constructor.mockedDispatchResolver(this, resolve, reject);
    });
  }
}

/**
 * Factory which creates an instance of MockedPipelineRequest.
 *
 * @param {function} cb Resolver callback.
 * @return {MockedPipelineRequest}
 */
export const mockedPipelineRequestFactory = cb =>
  class extends MockedPipelineRequest {
    /**
     * Getter for custom mocked resolver.
     * @return {*}
     */
    static get mockedDispatchResolver() {
      return cb;
    }
  };

