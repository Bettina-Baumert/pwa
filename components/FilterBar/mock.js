import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import { GRID_VIEW } from '../../pages/Category/constants';

const mockedStoreDefault = {
  filter: {
    activeFilters: [],
    availableFilters: {},
    activeHash: null,
    temporaryFilters: {},
  },
  ui: {
    categoryPage: {
      viewMode: GRID_VIEW,
    },
  },
  history: {
    queryParams: {
      sort: 'desc',
    },
  },
};

const mockedStoreAllSelected = {
  ...mockedStoreDefault,
  filter: {
    ...mockedStoreDefault.filter,
    activeFilters: [
      {
        filters: {
          SingleSelect: {
            type: 'single_select',
            label: 'Single selection',
            value: 'SingleSelected',
          },
          MultiSelect: {
            type: 'multiselect',
            label: 'MultiSelect',
            values: ['one', 'two'],
          },
          Range: {
            type: 'range',
            label: 'Range',
            minimum: 1.1,
            maximum: 99.99,
          },
        },
      },
    ],
  },
};

/**
 * Returns a store with default state.
 * @returns {Object}
 */
export function getDefaultStore() {
  return configureStore([thunk])(mockedStoreDefault);
}
/**
 * Returns a store with state with all filter types selected.
 * @returns {Object}
 */
export function getStoreWithSelectedFilters() {
  return configureStore([thunk])(mockedStoreAllSelected);
}
