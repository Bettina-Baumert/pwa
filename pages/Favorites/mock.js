/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export const mockedEmptyState = {
  product: {
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: [{}],
    },
  },
};
