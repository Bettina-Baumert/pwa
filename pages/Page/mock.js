const widgetRoute = {
  id: 'd82ea770-5627-4d1e-a301-a06bbe0d5cbd',
  params: {
    pageId: 'cms_test',
  },
  pathname: '/page/cms_test',
  pattern: '/page/:pageId',
  query: {},
  state: {},
};

const homeRoute = {
  id: 'a4bf3a32-0ed8-4138-bb94-098f32f3aeb3',
  params: {},
  pathname: '/',
  pattern: '/',
  query: {},
  state: {},
};

const routerState = {
  router: {
    routing: false,
    stack: [
      { ...homeRoute },
      { ...widgetRoute },
    ],
  },
};

const widgetsInitialState = {
  page: {},
  ...routerState,
};

export {
  widgetsInitialState,
};