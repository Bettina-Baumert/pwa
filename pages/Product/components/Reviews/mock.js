const mockedProduct = {
  productId: 'foo',
};

const mockedProductWithRating = {
  productId: 'foo',
  rating: {
    average: 50,
    count: 4,
  },
};

export const mockedStateWithAll = {
  product: {
    currentProduct: mockedProductWithRating,
    productsById: {
      foo: {
        productData: mockedProductWithRating,
      },
    },
    reviewsByProductId: {
      foo: {
        reviews: [
          {
            author: '',
            date: '2017-09-06T12:38:51.000Z',
            rate: 100,
            title: '',
            review: 'No Name and Title Lorem ipsum dolor sit amet, con… takimata sanctus est Lorem ipsum dolor sit amet.',
          },
          {
            author: 'username123',
            date: '2017-09-06T12:37:40.000Z',
            rate: 40,
            title: 'Test review 2 ',
            review: 'Lorem ipsum dolor sit amet, consetetur sadipscing … takimata sanctus est Lorem ipsum dolor sit amet.',
          },
          {
            author: 'Carina Hoffmann',
            date: '2017-09-06T12:30:23.000Z',
            rate: 60,
            title: '',
            review: 'Test review',
          },
          {
            author: '',
            date: '2017-09-06T12:30:23.000Z',
            rate: 20,
            title: '',
            review: '',
          },
        ],
      },
    },
  },
};

export const mockedStateWithProductOnly = {
  product: {
    currentProduct: mockedProduct,
    productsById: {
      foo: {
        productData: mockedProduct,
      },
    },
    reviewsByProductId: {},
  },
};
