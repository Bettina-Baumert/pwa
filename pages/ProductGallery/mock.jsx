import React from 'react';
import PropTypes from 'prop-types';

export const MockedComponent = class extends React.Component {
  static propTypes = {
    children: PropTypes.node,
  };
  static defaultProps = {
    children: null,
  };
  /**
   * Renders mocked view.
   * @return {JSX}
   */
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
};
