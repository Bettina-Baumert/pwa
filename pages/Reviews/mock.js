/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import {
  mockedStateWithProductOnly,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithProductOnly,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
