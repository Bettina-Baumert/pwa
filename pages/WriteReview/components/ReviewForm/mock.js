/**
 * Creates mocked review object.
 * @param {string|number} id Anything that can be an id.
 * @return {Object}
 */
export const mockReview = id => ({
  id,
  author: 'Jane Doe',
  date: '2017-09-06T12:38:51.000Z',
  rate: 100,
  title: 'Review with more than 255 characters Review with more than 255 ' +
  'charactersReview with more than 255 charactersReview with ' +
  'more than 255 charactersReview with more than 255 charactersReview with ' +
  'more than 255 charactersReview with more than 255 characters ',
  review: 'Lorem ipsum dolor sit amet, con… takimata sanctus est Lorem ipsum ' +
  'dolor sit amet.Lorem ipsum dolor sit amet, con… takimata sanctus est Lorem' +
  ' ipsum dolor sit amet.Lorem ipsum dolor sit amet, con… takimata sanctus est' +
  ' Lorem ipsum dolor sit amet.Lorem ipsum dolor sit amet, con… takimata sanctus' +
  ' est Lorem ipsum dolor sit amet.',
  productId: 'foo',
});

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedStateWithoutReview = {
  product: {
    currentProduct: {
      productId: 'foo',
    },
    productsById: {
      foo: {
        productData: {
          productId: 'foo',
        },
      },
    },
  },
  reviews: {
    reviewsById: {},
    userReviewsByProductId: {},
  },
  history: {
    pathname: '/item/666f6f/write_review',
  },
  user: {
    data: {
      isFetching: false,
      firstName: 'Jane',
      lastName: 'Doe',
    },
  },
  view: {
    isLoading: false,
  },
};

export const mockedStateWithInvalidReview = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsById: {
      1: mockReview(1),
    },
    userReviewsByProductId: {
      foo: { review: 1 },
    },
  },
};

export const mockedStateWithReview = {
  ...mockedStateWithInvalidReview,
  reviews: {
    reviewsById: {
      1: {
        ...mockedStateWithInvalidReview.reviews.reviewsById[1],
        title: 'Lorem ipsum dolor sit amet',
        review: 'takimata sanctus est Lorem ipsum',
      },
    },
  },
};
