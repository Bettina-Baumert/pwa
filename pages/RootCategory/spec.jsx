import React from 'react';
import { mount } from 'enzyme';
import { initialCategoryState, categoryState } from '@shopgate/pwa-common-commerce/category/mock';
import { UnwrappedRootCategory as RootCategory } from './index';

jest.mock('Components/List/components/Item', () => {
  /**
   * Renders a ListItem mock.
   * @return {JSX}
   */
  const ListItem = () => <div />;
  return ListItem;
});

jest.mock('Components/View');
let store;

const results = [
  [{
    type: 'NAVIGATE',
    params: {
      action: 'PUSH',
      pathname: '/category/6d656e',
      state: { title: 'Men' },
    },
  }],
];

/**
 * Creates component
 * @param {Object} props Component props.
 * @return {ReactWrapper}
 */
const createComponent = props => mount(<RootCategory {...props} />);

describe('<RootCategory> page', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('should render', () => {
    const categories = categoryState.category.categoriesById;
    const props = {
      open: true,
      categories: [categories.men, categories.women],
    };
    const component = createComponent(props);
    expect(component).toMatchSnapshot();
    expect(component.find('CategoryList').length).toEqual(1);
    expect(component.find('ListItem').length).toEqual(4);
  });

  it('should render empty', () => {
    const props = {
      open: true,
      categories: null,
    };
    const component = createComponent(props);
    expect(component).toMatchSnapshot();
    expect(component.find('CategoryList').length).toEqual(0);
    expect(component.find('ListItem').length).toEqual(0);
  });

  it.skip('should navigate to subcategory', () => {
    const wrapper = createComponent(categoryState);

    wrapper.find('CategoryList').find('Connect(Link)').at(0).simulate('click');
    wrapper.update();
    expect(store.getActions()).toEqual(results[0]);
  });
});
