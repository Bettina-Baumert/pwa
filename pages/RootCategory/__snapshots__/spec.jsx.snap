// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RootCategory> page should render 1`] = `
<RootCategory
  categories={
    Array [
      Object {
        "children": Array [],
        "childrenCount": 0,
        "childrenSort": "import",
        "description": "",
        "expires": 9999999999999,
        "externalUrl": "http://material-fashion.myshopify.com/collections/men",
        "id": "men",
        "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346609@2x.jpg",
        "isFetching": false,
        "name": "Men",
        "parent": Object {
          "id": null,
          "name": "Default- category",
        },
        "path": "Men",
        "productCount": 6,
        "sort": 0,
      },
      Object {
        "children": Array [],
        "childrenCount": 1,
        "childrenSort": "import",
        "description": "",
        "expires": 9999999999999,
        "externalUrl": "http://material-fashion.myshopify.com/collections/women",
        "id": "women",
        "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346610@2x.jpg",
        "isFetching": false,
        "name": "Women",
        "parent": Object {
          "id": null,
          "name": "Default- category",
        },
        "path": "Women",
        "productCount": 6,
        "sort": 0,
      },
    ]
  }
  open={true}
>
  <MockedView>
    <div>
      <CategoryList
        categories={
          Array [
            Object {
              "children": Array [],
              "childrenCount": 0,
              "childrenSort": "import",
              "description": "",
              "expires": 9999999999999,
              "externalUrl": "http://material-fashion.myshopify.com/collections/men",
              "id": "men",
              "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346609@2x.jpg",
              "isFetching": false,
              "name": "Men",
              "parent": Object {
                "id": null,
                "name": "Default- category",
              },
              "path": "Men",
              "productCount": 6,
              "sort": 0,
            },
            Object {
              "children": Array [],
              "childrenCount": 1,
              "childrenSort": "import",
              "description": "",
              "expires": 9999999999999,
              "externalUrl": "http://material-fashion.myshopify.com/collections/women",
              "id": "women",
              "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346610@2x.jpg",
              "isFetching": false,
              "name": "Women",
              "parent": Object {
                "id": null,
                "name": "Default- category",
              },
              "path": "Women",
              "productCount": 6,
              "sort": 0,
            },
          ]
        }
      >
        <List
          hasImages={false}
        >
          <List
            className=""
          >
            <ul
              className="css-gkw1j5 "
            >
              <ListItem
                className="css-1nftllw css-1ym3l2n"
                isSelected={false}
                key="child-0/.$men"
              >
                <li
                  className="css-1p5q8vq css-1nftllw css-1ym3l2n css-16gj3o1"
                >
                  <div
                    className="css-1ckzvk7"
                  >
                    <Portal
                      key="men"
                      name="category.item"
                      props={
                        Object {
                          "categoryId": "men",
                        }
                      }
                    >
                      <ListItem
                        link="/category/6d656e"
                        title="Men"
                      >
                        <div />
                      </ListItem>
                    </Portal>
                  </div>
                </li>
              </ListItem>
              <ListItem
                className="css-1nftllw"
                isSelected={false}
                key="child-1/.$women"
              >
                <li
                  className="css-1p5q8vq css-1nftllw css-16gj3o1"
                >
                  <div
                    className="css-1ckzvk7"
                  >
                    <Portal
                      key="women"
                      name="category.item"
                      props={
                        Object {
                          "categoryId": "women",
                        }
                      }
                    >
                      <ListItem
                        link="/category/776f6d656e"
                        title="Women"
                      >
                        <div />
                      </ListItem>
                    </Portal>
                  </div>
                </li>
              </ListItem>
            </ul>
          </List>
        </List>
      </CategoryList>
    </div>
  </MockedView>
</RootCategory>
`;

exports[`<RootCategory> page should render empty 1`] = `
<RootCategory
  categories={null}
  open={true}
>
  <MockedView>
    <div />
  </MockedView>
</RootCategory>
`;
